validphys2/src/validphys/config.py

```diff
@@ -1305,7 +1314,7 @@ def produce_rules(
         try:
             rule_list = [
                 Rule(
-                    initial_data=rule,
+                    initial_data=rule.to_dict(),
```

Comment:
I think that instead of passing `.to_dict` here, it is better to do, in the class `Rule`, 

```python
# For compatibility with legacy code that passed a dictionary
# TODO: make sure there are no calls left to `Rules` with a dictionary input
initial_data = initial_data.to_dict()
```

(you might want to cover this with a `isinstance` or `try-except`)